apps/backend/src/participant/participant.controller.ts

```diff
@@ -43,10 +55,23 @@ export class ParticipantController {
   }
 
   @Delete(':participant')
+  @UseGuards(OptionalAuthGuard)
   async delete(
     @Param('poll', ObjectIdPipe) poll: Types.ObjectId,
     @Param('participant', ObjectIdPipe) participant: Types.ObjectId,
+    @Headers('Participant-Token') token: string,
+    @AuthUser() user?: UserToken,
   ): Promise<ReadParticipantDto | null> {
+    const [participantDoc, pollDoc] = await Promise.all([
+      this.pollService.findParticipant(participant),
+      this.pollService.find(poll),
+    ]);
+    if (!participantDoc || !pollDoc) {
+      throw new NotFoundException('Poll or participant not found');
+    }
+    if (participant.toString() !== token && !this.pollService.isAdmin(pollDoc, token, user?.sub)) {
```

Comment:
This doesnt make sense, participant is the Id, you need to compare token against participantDoc token

Reply:
Was already fixed in [d364cc3](https://github.com/Morphclue/apollusia/pull/228/commits/d364cc3bb9aa39c90ffb5044db5fb25845297051).